### DISCUSSION TITLE
After upgrade, best way to get rid of no longer available entities?

### DISCUSSION THREAD

**Discussion by fordcrews** [Blackymas/NSPanel_HA_Blueprint#1070] [General]
After upgrading to beta, then 4.0, I have a bunch of entities no longer available.  NSPanel event, Setting Entity,  page timeout, Relay 1/2 local fall back, sleep mode, etc.

What is the best way to get rid of these?

Delete them and readd them in esphome?

**Comment by edwardtfn** [comment]
You can delete those.

**Reply by fordcrews** [reply]
But where, in the ui you click them, the delete is grayed out because they are unavailable.   Only way I could see to delete them was to remove the esphome device and readd it, which I always hesitate to do, because in the past I have had them with passwords, I didn't document and when I tried to re add them, I ended up having to reflash them ;-)

**Reply by edwardtfn** [reply]
I believe you will be able to delete those after a Home Assistant reboot.

**Reply by fordcrews** [reply]
That did it, I didn't think to reboot, and was hoping I had missed a homeassistant service or something like remove missing entities, I have 10 of these configured at the moment, it will hake me a half an hour to delete them one by one.    So I guess I will drop them and readd the entire device.

**Reply by edwardtfn** [reply]
> So I guess I will drop them and readd the entire device.

That should work also.

**Comment by edwardtfn** [comment]
Take a look at this: https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/1057#discussioncomment-7064904

**Reply by fordcrews** [reply]
I called myself searching, not sure how I missed this.

**Comment by DozoG** [comment]
I am leaving this short update note because not all of the new functions are mentioned in the manuals yet.

Updated two NSPanels from 3.3 to 4.0
Several items are no longer used (amongst those I used)
- Relay local Fallback (I think is default now)
- Page Timeout
- Sleep mode
Have been replaced by - Timeout Dim, - Timeout Page and - Timeout Sleep.
Which when set to zero will be ignored.
I had some automations in NodeRed that set the SleepMode after midnight, Those I will replace by setting timeout Sleep = 0 in the day and = 60 seconds after midnight.


Also added is wake-up page .. which is a great feature. One of my Panels is just used as a multi-light switch, which can handle 6 HomeAssistant light switches (so they easily fit on one button page) . I can now always get right to that page when I touch the display.


Great work and thanks to the developers.

**Reply by edwardtfn** [reply]
Thanks for sharing this.

In fact, the relay local fallback switches were used by the Blueprint to send the information to ESPHome, not intended to be changed directly by users, but in the end these settings have always been part of the Blueprint and this didn't changed.
It was possible to switch this oj the device page (the switches), but that was always being reset by the Blueprint everytime the panel boots up, which was far from intuitive.
The change on v4.0 was to replace those switches by a service call, so we reduce the risk of users changing those settings manually and the system not behaving as they wish.
I'm trying to clean up all those entities used only for internal communication between the Blueprint and ESPHome (as was the case also for those json-like sensors).

The fallback is not default. You still have to enable it if you want the panel to fallback to local relay when Home Assistant or Wi-Fi is disabled, just like v3.4.
What is also new now is that if you select relay 1 for left button or relay 2 to right button (and only on these situations), instead of sensing the button press from the panel to the Blueprint, having the Blueprint processing the command and calling back the panel to toggle the relay, now all of this is happening internally in the panel, saving network flow, processing but, more important, reducing the response time and enabling the operation when Wi-Fi is out, which is basically saying that in this specific case, fallback is always applied.
